lib/IDE/CodeCompletion.cpp

```diff
@@ -2481,16 +2481,17 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
         // What's left is the result type.
         if (ResultType->isVoid()) {
           OS << "Void";
-        } else if (!IsImplicitlyCurriedInstanceMethod
-                   && FD->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>()) {
+        } else {
           // As we did with parameters in addParamPatternFromFunction,
           // for regular methods we'll print '!' after implicitly
           // unwrapped optional results.
-          auto ObjectType = ResultType->getOptionalObjectType();
-          OS << ObjectType->getStringAsComponent();
```

Comment:
I couldn't find a reproducer for this crash. But we have several logs crashed in here. It means, result type is *not* optional even when the function decl has `ImplicitlyUnwrappedOptionalAttr`.
@rudkx Are you able to come up with a possibility that causes this situation?

Reply:
The only thing I can think of is that this is related to the adjustment of `FunctionType` earlier in the function.